### PR TITLE
Switch to the fast broadcast implementation

### PR DESCRIPTION
**Pull request by YingboMa**
Macro `@..` is the fast broadcast implementation which assumes that all arrays are non-extruded, so that LLVM can always do a runtime alias check and vectorize.

### CODE REVIEW

**Comment by YingboMa** [comment]
```julia
julia> using OrdinaryDiffEq, BenchmarkTools

julia> using OrdinaryDiffEq: perform_step!, initialize!

julia> prob10 = ODEProblem((du, u, p, t)->copyto!(du, u),ones(10),(0.0,10.0));

julia> prob100 = ODEProblem((du, u, p, t)->copyto!(du, u),ones(100),(0.0,10.0));

julia> prob1000 = ODEProblem((du, u, p, t)->copyto!(du, u),ones(1000),(0.0,10.0));

julia> integ10 = init(prob10, Tsit5()); integ100 = init(prob100, Tsit5()); integ1000 = init(prob1000, Tsit5());

julia> initialize!(integ10, integ10.cache); initialize!(integ100, integ100.cache); initialize!(integ1000, integ1000.cache);

julia> @btime perform_step!($integ10,   $integ10.cache, false) # PR
  251.051 ns (0 allocations: 0 bytes)

julia> @btime perform_step!($integ100,  $integ100.cache, false) # PR
  645.738 ns (0 allocations: 0 bytes)

julia> @btime perform_step!($integ1000, $integ1000.cache, false) # PR
  5.743 μs (0 allocations: 0 bytes)

julia> @btime perform_step!($integ10,   $integ10.cache, false) # Master
  220.204 ns (0 allocations: 0 bytes)

julia> @btime perform_step!($integ100,  $integ100.cache, false) # Master
  620.471 ns (0 allocations: 0 bytes)

julia> @btime perform_step!($integ1000, $integ1000.cache, false) # Master
  5.754 μs (0 allocations: 0 bytes)
```
Broadcast has a constant overhead on the order of 1-2 ns (there are 7 broadcasts in adaptive Tsit5).

**Comment by YingboMa** [comment]
With https://github.com/JuliaDiffEq/DiffEqBase.jl/pull/204, I got
```julia
using OrdinaryDiffEq, BenchmarkTools
prob = ODEProblem((du, u, p, t)->copyto!(du, u),ones(1000),(0.0,10.0))
integ = init(prob, Tsit5());
@btime OrdinaryDiffEq.perform_step!($integ, $(integ.cache)) # bc
@btime OrdinaryDiffEq.perform_step!($integ, $(integ.cache)) # loop
#=
julia> @btime OrdinaryDiffEq.perform_step!($integ, $(integ.cache)) # bc
  4.122 μs (0 allocations: 0 bytes)

julia> @btime OrdinaryDiffEq.perform_step!($integ, $(integ.cache)) # loop
  4.129 μs (0 allocations: 0 bytes)
=#
integ = init(prob, Vern9());
@btime OrdinaryDiffEq.perform_step!($integ, $(integ.cache)) # bc
@btime OrdinaryDiffEq.perform_step!($integ, $(integ.cache)) # loop
#=
julia> @btime OrdinaryDiffEq.perform_step!($integ, $(integ.cache)) # bc
  11.411 μs (0 allocations: 0 bytes)

julia> @btime OrdinaryDiffEq.perform_step!($integ, $(integ.cache)) # loop
  20.662 μs (0 allocations: 0 bytes)
=#
```
.

I think that it is safe to say that there is no regression :-)

**Comment by YingboMa** [comment]
This PR needs https://github.com/JuliaDiffEq/DiffEqBase.jl/pull/204

File: src/nlsolve/utils.jl [hunk 1]
```diff
@@ -50,7 +50,7 @@ function qradd!(Q::AbstractMatrix, R::AbstractMatrix, v::AbstractVector, k::Int)
   @inbounds begin
     d = norm(v)
     R[k, k] = d
-    @. Q[:, k] = v / d
+    @.. @view(Q[:, k]) = v / d
```
Comment (ChrisRackauckas) [inline src/nlsolve/utils.jl]: oh lol. That might effect timings.

**Comment by ChrisRackauckas** [comment]
Nord fails.

**Pull request merged**